lib/OpenQA/CLI/api.pm

```diff
@@ -26,16 +26,30 @@ sub command ($self, @args) {
       'd|data=s' => \$data,
       'f|form' => \my $form,
       'j|json' => \my $json,
+      'H|hook-scripts=s' => \my $hooksrc,
       'param-file=s' => \my @param_file,
       'r|retries=i' => \my $retries,
       'X|method=s' => \(my $method = 'GET');
 
     @args = $self->decode_args(@args);
     die $self->usage unless my $path = shift @args;
 
-    $data = path($data_file)->slurp if $data_file;
-    my @data = ($data);
-    my $params = $form ? decode_json($data) : $self->parse_params(\@args, \@param_file);
+    my $params;
+    my @data;
+    if ($hooksrc) {
+	# if $hooksrc is a json  
+        my $hookdata = path("hook/$hooksrc.json")->slurp;
```

Comment:
This is very unintuitive. Let's keep it simple and not "surround" the specified path with `hook/`…`.json`.

Just invoke the specified hook script reading its stdout. Abort with a clear error if it exited with a non-zero return code. Otherwise, parse the output as JSON. There's *no* reason to involve an intermediate `.json` file that must be placed according to some conventions.

Note that you can use the quote like operator `qx(…)` in Perl to invoke a script and get its output. When using it, have a close look at [the documentation](https://perldoc.perl.org/perlop#Quote-Like-Operators) because error handling is a bit involved (but should be done correctly to get a clear error message in any case).